packages/editor/src/components/document-outline/index.js

```diff
@@ -147,35 +201,72 @@ export default function DocumentOutline( {
 
 	const prevHeadingLevelRef = useRef( 1 );
 
-	const headings = useMemo(
-		() => computeOutlineHeadings( blocks ),
+	const outlineElements = useMemo(
+		() => computeOutlineElements( blocks ),
 		[ blocks ]
 	);
 
-	if ( headings.length < 1 ) {
-		return (
-			<div className="editor-document-outline has-no-headings">
-				<EmptyOutlineIllustration />
-				<p>
-					{ __(
-						'Navigate the structure of your document and address issues like empty or incorrect heading levels.'
-					) }
-				</p>
-			</div>
-		);
-	}
+	const headings = outlineElements.filter(
+		( item ) => item.type === 'heading'
+	);
+	const mainElements = outlineElements.filter(
+		( item ) => item.type === 'main'
+	);
+
+	const isShowingNotice = useSelect(
+		( select ) =>
+			!! select( noticesStore )
+				.getNotices()
+				.find(
+					( notice ) => notice.id === 'missing-main-html-element'
+				),
+		[]
+	);
+
+	useEffect( () => {
+		if (
+			! isNoticeDismissed &&
+			! isShowingNotice &&
+			isShowingTemplate &&
+			mainElements.length === 0
+		) {
+			createNotice(
+				'warning',
+				__(
+					'The main HTML element is missing. Select the block that contains your most important content and add the main HTML element in the Advanced panel.'
```

Comment:
`createNotice ` can not include the link to the HTML reference, so we end up with two very similar text strings.